File: iPad/VerBank-CSTSv3-IOSClientApi/CSTS3Comm/comm/BalanceRate.m
//
//  BalanceRate.m
//  VerBank-CSTSv3-IOSClientApi
//
//  Created By Zhanglei on 15/6/29.
//  Copyright (c) 2015年 allone. All rights reserved.
//

#import "BalanceRate.h"

static NSString * jsonId = @"BalanceRate";

static NSString * instrument = @"1";
static NSString * tradeDay = @"2";
static NSString * Cur1 = @"3";
static NSString * Cur2 = @"4";
static NSString * Bid = @"5";
static NSString * Ask = @"6";

@implementation BalanceRate
- (id)init{
    self = [super init];
    if(self != nil)
    {
        [super setEntry:@"jsonId" entry:jsonId];
    }
    return self;
}

jsonImplementionString(Instrument,  instrument)
jsonImplementionString(TradeDay,    tradeDay)
jsonImplementionString(Cur1,        Cur1)
jsonImplementionString(Cur2,        Cur2)
jsonImplementionDouble(Bid,         Bid)
jsonImplementionDouble(Ask,         Ask)

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/CSTS3/comm/struct/BalanceRate.java
package jedi.ex01.CSTS3.comm.struct;


public class BalanceRate extends allone.json.AbstractJsonData {

	public static final String jsonId = "3";

	public static final String instrument = "1";
	public static final String cur1 = "2";
	public static final String cur2 = "3";
	public static final String bid = "4";
	public static final String ask = "5";
	public static final String tradeDay = "6";

	public BalanceRate(){
		super();
		setEntry("jsonId", jsonId);
	}

	public String getInstrument() {
		try {
			String data=getEntryString(BalanceRate.instrument);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setInstrument(String instrument) {
		setEntry(BalanceRate.instrument, instrument);
	}

	public String getCur1() {
		try {
			String data=getEntryString(BalanceRate.cur1);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setCur1(String cur1) {
		setEntry(BalanceRate.cur1, cur1);
	}

	public String getCur2() {
		try {
			String data=getEntryString(BalanceRate.cur2);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setCur2(String cur2) {
		setEntry(BalanceRate.cur2, cur2);
	}

	public double getBid() {
		try {
			double data=getEntryDouble(BalanceRate.bid);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setBid(double bid) {
		setEntry(BalanceRate.bid, bid);
	}

	public double getAsk() {
		try {
			double data=getEntryDouble(BalanceRate.ask);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setAsk(double ask) {
		setEntry(BalanceRate.ask, ask);
	}

	public String getTradeDay() {
		try {
			String data=getEntryString(BalanceRate.tradeDay);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setTradeDay(String tradeDay) {
		setEntry(BalanceRate.tradeDay, tradeDay);
	}


}